Still one nagging warning on gcc-9.3

@KineticTheory Still one nagging warning on gcc-9.3:  😢 

`src/rng/test/kat_cpp.cpp:230:5: warning: this use of "defined" may not be portable`
 ` 230 | #if R123_USE_PHILOX_64BIT`

_Originally posted by @keadyk in https://github.com/lanl/Draco/issues/999#issuecomment-751147891_

--- a/src/c4/test/CMakeLists.txt
+++ b/src/c4/test/CMakeLists.txt
@@ -55,6 +55,9 @@ set( special_tests
 if( DEFINED ENV{TRAVIS} )
   # TRAVIS has severe limits on number of ranks/threads
   set( omp_test_pe_list "1" )
+elseif( CODE_COVERAGE )
+  # The 1 MPI rank version gets confused when running under code-coverage configurations.
+  set( omp_test_pe_list "2" )
 else()
   set( omp_test_pe_list "1;2" )
 endif()

--- a/src/rng/test/CMakeLists.txt
+++ b/src/rng/test/CMakeLists.txt
@@ -8,7 +8,8 @@
 project( rng_test C CXX )
 
 if( CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9.0 )
-  toggle_compiler_flag( TRUE "-Wno-expansion-to-defined" "CXX" "DEBUG")
+  string(APPEND CMAKE_C_FLAGS " -Wno-expansion-to-defined")
+  string(APPEND CMAKE_CXX_FLAGS " -Wno-expansion-to-defined")
 endif()
 
 #--------------------------------------------------------------------------------------------------#

--- a/src/rng/test/CMakeLists.txt
+++ b/src/rng/test/CMakeLists.txt
@@ -7,6 +7,10 @@
 #--------------------------------------------------------------------------------------------------#
 project( rng_test C CXX )
 
+if( CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9.0 )
+  toggle_compiler_flag( TRUE "-Wno-expansion-to-defined" "CXX" "DEBUG")
+endif()
+
 #--------------------------------------------------------------------------------------------------#
 # Source files
 #--------------------------------------------------------------------------------------------------#

